### PR TITLE
resource/cognito_user_pool_client: support USER_PASSWORD_AUTH for explicit_auth_flows

### LINKED ISSUE #3397: AWS Cognito user pool client does not allow USER_PASSWORD_AUTH as explicit_auth_flows
Hi,

It seems AWS Cognito app clients allows one more configuration option now. It's not possible to set USER_PASSWORD_AUTH for an app client with Terraform.

### Terraform Version
Terraform v0.11.3

### Affected Resource(s)
- aws_cognito_user_pool_client

### Terraform Configuration Files
```hcl
resource "aws_cognito_user_pool_client" "main" {
  name = "app"

  user_pool_id = "${aws_cognito_user_pool.main.id}"
  explicit_auth_flows = ["ADMIN_NO_SRP_AUTH", "USER_PASSWORD_AUTH"]
}
```

### Expected Behavior
Create an app client with desired options

### Actual Behavior
<pre>
Error: aws_cognito_user_pool_client.main: expected explicit_auth_flows.1 to be one of [ADMIN_NO_SRP_AUTH CUSTOM_AUTH_FLOW_ONLY], got USER_PASSWORD_AUTH
</pre>

### PR DESCRIPTION
**Pull request by loivis**
Fix #3397 
Documentation: https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_CreateUserPoolClient.html#CognitoUserPools-CreateUserPoolClient-request-ExplicitAuthFlows

### CODE REVIEW

File: aws/resource_aws_cognito_user_pool_client.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ func resourceAwsCognitoUserPoolClient() *schema.Resource {
 					ValidateFunc: validation.StringInSlice([]string{
 						cognitoidentityprovider.ExplicitAuthFlowsTypeAdminNoSrpAuth,
 						cognitoidentityprovider.ExplicitAuthFlowsTypeCustomAuthFlowOnly,
+						cognitoidentityprovider.ExplicitAuthFlowsTypeUserPasswordAuth,
```
Comment (Ninir) [inline aws/resource_aws_cognito_user_pool_client.go]: Could you add an acctest for that so that we ensure there won't be regression in the future?

Comment (loivis) [reply aws/resource_aws_cognito_user_pool_client.go]: I thought about acctest before committing the change but not sure what to test actually. `USER_PASSWORD_AUTH` is a new type of `ExplicitAuthFlows`.

Comment (bflad) [reply aws/resource_aws_cognito_user_pool_client.go]: My general opinion is if the SDK is providing the constant, that means AWS has added it to their acceptable values in their generated API model, and we don't need to provide any acceptance testing if its just updating a validation check. 😄

Comment (Ninir) [reply aws/resource_aws_cognito_user_pool_client.go]: I recall, when working on multiple Cognito ressources, that there was a specificity when using USER_PASSWORD_AUTH... hence my request :)
Adding the value in `testAccAWSCognitoUserPoolClientConfig_allFields` is fine as it covers the use-case.

Comment (loivis) [reply aws/resource_aws_cognito_user_pool_client.go]: Added all values of `explicit_auth_flows` in acctest. I'm not clear about the user-case but hope to get your point of change.

**Comment by loivis** [comment]
```
⎇  make testacc TEST=./aws TESTARGS='-run=TestAccAWSCognitoUserPoolClient_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSCognitoUserPoolClient_ -timeout 120m
=== RUN   TestAccAWSCognitoUserPoolClient_basic
--- PASS: TestAccAWSCognitoUserPoolClient_basic (21.71s)
=== RUN   TestAccAWSCognitoUserPoolClient_allFields
--- PASS: TestAccAWSCognitoUserPoolClient_allFields (21.19s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	42.948s
```

File: aws/resource_aws_cognito_user_pool_client.go (hunk 1 above)
Comment (Ninir) [reply aws/resource_aws_cognito_user_pool_client.go]: Cannot recall of the need for it right now... it was not that important, but will keep you updated if it appears to me again.

**Comment by bflad** [comment]
This has been released in [version 1.10.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1100-february-24-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**